onnxruntime/core/providers/openvino/openvino_provider_factory.cc

```diff
@@ -30,6 +30,10 @@ struct OpenVINOProviderFactory : IExecutionProviderFactory {
         so_epctx_embed_mode_(so_epctx_embed_mode) {
     device_type_ = (device_type == nullptr) ? "" : device_type;
     cache_dir_ = (cache_dir == nullptr) ? "" : cache_dir;
+    if (cache_dir != nullptr) {
+      free(const_cast<void*>(static_cast<const void*>(cache_dir)));
```

Comment:
>free( [](http://example.com/codeflow?start=6&length=5)

Are we sure that cache_dir is allocated by malloc? We use new/delete through the codebase.

Reply:
i think because they use strdup to allocate the memory
https://github.com/microsoft/onnxruntime/blob/953ea3f0b657ed0d232850baf56dd4cab2183e60/onnxruntime/core/providers/openvino/openvino_provider_factory.cc#L321

Reply:
intel will fix in follow-up PR